diff --git a/node_modules/react-native-screens/ios/bottom-tabs/RNSTabBarController.mm b/node_modules/react-native-screens/ios/bottom-tabs/RNSTabBarController.mm
index 1c70eec..375cba4 100644
--- a/node_modules/react-native-screens/ios/bottom-tabs/RNSTabBarController.mm
+++ b/node_modules/react-native-screens/ios/bottom-tabs/RNSTabBarController.mm
@@ -19,6 +19,15 @@
     _tabBarAppearanceCoordinator = [RNSTabBarAppearanceCoordinator new];
     _tabsHostComponentView = nil;
 
+#if !TARGET_OS_TV
+    if (@available(iOS 18.0, *)) {
+      if ([[UIDevice currentDevice] userInterfaceIdiom] == UIUserInterfaceIdiomPad) {
+        [self setMode:UITabBarControllerModeTabSidebar]; // Enable the sidebar
+        self.sidebar.hidden = YES; // Hide it by default
+      }
+    }
+#endif
+
 #if !RCT_NEW_ARCH_ENABLED
     _isControllerFlushBlockScheduled = NO;
 #endif // !RCT_NEW_ARCH_ENABLED
