diff --git a/node_modules/react-native-screens/ios/bottom-tabs/RNSBottomTabsScreenComponentView.mm b/node_modules/react-native-screens/ios/bottom-tabs/RNSBottomTabsScreenComponentView.mm
index 9fd4afc..4fd41d9 100644
--- a/node_modules/react-native-screens/ios/bottom-tabs/RNSBottomTabsScreenComponentView.mm
+++ b/node_modules/react-native-screens/ios/bottom-tabs/RNSBottomTabsScreenComponentView.mm
@@ -177,7 +177,7 @@ RNS_IGNORE_SUPER_CALL_END
     _tabBarItemBadgeBackgroundColor = RCTUIColorFromSharedColor(newComponentProps.tabBarItemBadgeBackgroundColor);
     // Note that this will prevent default color from being set.
     // TODO: support default color by setting nil here.
-    NSLog(@"TabsScreen [%ld] update badgeColor to %@", self.tag, _tabBarItemBadgeBackgroundColor);
+    // NSLog(@"TabsScreen [%ld] update badgeColor to %@", self.tag, _tabBarItemBadgeBackgroundColor);
     tabItemNeedsAppearanceUpdate = YES;
   }
 
@@ -297,8 +297,8 @@ RNS_IGNORE_SUPER_CALL_END
 - (void)updateLayoutMetrics:(const facebook::react::LayoutMetrics &)layoutMetrics
            oldLayoutMetrics:(const facebook::react::LayoutMetrics &)oldLayoutMetrics
 {
-  NSLog(
-      @"TabScreen [%ld] updateLayoutMetrics: %@", self.tag, NSStringFromCGRect(RCTCGRectFromRect(layoutMetrics.frame)));
+  // NSLog(
+  //     @"TabScreen [%ld] updateLayoutMetrics: %@", self.tag, NSStringFromCGRect(RCTCGRectFromRect(layoutMetrics.frame)));
   [super updateLayoutMetrics:layoutMetrics oldLayoutMetrics:oldLayoutMetrics];
 }
 
@@ -311,13 +311,13 @@ RNS_IGNORE_SUPER_CALL_END
 
 - (void)mountChildComponentView:(UIView<RCTComponentViewProtocol> *)childComponentView index:(NSInteger)index
 {
-  NSLog(@"TabScreen [%ld] mount [%ld] at %ld", self.tag, childComponentView.tag, index);
+  // NSLog(@"TabScreen [%ld] mount [%ld] at %ld", self.tag, childComponentView.tag, index);
   [super mountChildComponentView:childComponentView index:index];
 }
 
 - (void)unmountChildComponentView:(UIView<RCTComponentViewProtocol> *)childComponentView index:(NSInteger)index
 {
-  NSLog(@"TabScreen [%ld] unmount [%ld] from %ld", self.tag, childComponentView.tag, index);
+  // NSLog(@"TabScreen [%ld] unmount [%ld] from %ld", self.tag, childComponentView.tag, index);
   [super unmountChildComponentView:childComponentView index:index];
 }
 
diff --git a/node_modules/react-native-screens/ios/bottom-tabs/RNSTabBarController.mm b/node_modules/react-native-screens/ios/bottom-tabs/RNSTabBarController.mm
index 8c2b0c7..5530441 100644
--- a/node_modules/react-native-screens/ios/bottom-tabs/RNSTabBarController.mm
+++ b/node_modules/react-native-screens/ios/bottom-tabs/RNSTabBarController.mm
@@ -34,7 +34,7 @@
 
 - (void)tabBar:(UITabBar *)tabBar didSelectItem:(UITabBarItem *)item
 {
-  NSLog(@"TabBar: %@ didSelectItem: %@", tabBar, item);
+  // NSLog(@"TabBar: %@ didSelectItem: %@", tabBar, item);
 }
 
 #pragma mark-- Signals
@@ -74,12 +74,12 @@
 
 - (void)reactMountingTransactionWillMount
 {
-  NSLog(@"TabBarCtrl mountintTransactionWillMount");
+  // NSLog(@"TabBarCtrl mountintTransactionWillMount");
 }
 
 - (void)reactMountingTransactionDidMount
 {
-  NSLog(@"TabBarCtrl mountintTransactionDidMount running updates");
+  // NSLog(@"TabBarCtrl mountintTransactionDidMount running updates");
   [self updateReactChildrenControllersIfNeeded];
   [self updateSelectedViewControllerIfNeeded];
   [self updateTabBarAppearanceIfNeeded];
@@ -96,7 +96,7 @@
 
 - (void)updateReactChildrenControllers
 {
-  NSLog(@"TabBarCtrl updateReactChildrenControllers");
+  // NSLog(@"TabBarCtrl updateReactChildrenControllers");
   _needsUpdateOfReactChildrenControllers = false;
 
   if (_tabScreenControllers == nil) {
@@ -116,7 +116,7 @@
 
 - (void)updateSelectedViewController
 {
-  NSLog(@"TabBarCtrl updateSelectedViewController");
+  // NSLog(@"TabBarCtrl updateSelectedViewController");
   _needsUpdateOfSelectedTab = false;
 
 #if !defined(NDEBUG)
@@ -125,10 +125,12 @@
 
   RNSTabsScreenViewController *_Nullable selectedViewController = nil;
   for (RNSTabsScreenViewController *tabViewController in self.viewControllers) {
+    /*
     NSLog(
         @"Update Selected View Controller [%ld] isFocused %d",
         tabViewController.tabScreenComponentView.tag,
         tabViewController.tabScreenComponentView.isSelectedScreen);
+     */
     if (tabViewController.tabScreenComponentView.isSelectedScreen == true) {
       selectedViewController = tabViewController;
       break;
@@ -152,7 +154,7 @@
 
 - (void)updateTabBarAppearance
 {
-  NSLog(@"TabBarCtrl updateTabBarAppearance");
+  // NSLog(@"TabBarCtrl updateTabBarAppearance");
   _needsUpdateOfTabBarAppearance = false;
 
   [_tabBarAppearanceCoordinator updateAppearanceOfTabBar:[self tabBar]
diff --git a/node_modules/react-native-screens/ios/bottom-tabs/RNSTabsScreenViewController.mm b/node_modules/react-native-screens/ios/bottom-tabs/RNSTabsScreenViewController.mm
index 24cf45e..79872ac 100644
--- a/node_modules/react-native-screens/ios/bottom-tabs/RNSTabsScreenViewController.mm
+++ b/node_modules/react-native-screens/ios/bottom-tabs/RNSTabsScreenViewController.mm
@@ -54,7 +54,7 @@
 
 - (void)didMoveToParentViewController:(UIViewController *)parent
 {
-  NSLog(@"TabScreen [%ld] ctrl moved to parent: %@", self.tabScreenComponentView.tag, parent);
+  // NSLog(@"TabScreen [%ld] ctrl moved to parent: %@", self.tabScreenComponentView.tag, parent);
 
   if (parent == nil) {
     return;
